.github/workflows/workflow.yml

```diff
@@ -0,0 +1,77 @@
+name: Test and Deploy
+on:
+  pull_request:
+    branches: 
+      - master
+  push:
+    branches:
+      - master
+  create:
+    tags:
+      - '*'
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Black
+      run: |
+        pip install black
+        black --check --diff .
+
+  test-extension:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.7',  '3.8', 'pypy3']
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dev dependencies
+        run: |
+          set -xe
+          python -VV
+          python -m site
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install -r dev-requirements.txt
+      - name: Install Package
+        run: |
+          python -m pip install .
+      - name: Install test dependencies
+        run: |
+          python -m pip install -r test-requirements.txt
+      - name: Run Tests for ${{ matrix.python-version }}
+        run: |
+          cd tests
+          python -m pytest -vv
+  pypi-release:
+    needs: test-extension
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'refs/tags/') && github.repository_owner	== 'wpilibsuite'
```

Comment:
Should this if statement be only on the publish step - that way we can test building the wheel?

Reply:
I added building a wheel to the test job so it should be fine behind a guard in pypi-release.